### PR TITLE
Anagram: sorting approach

### PR DESCRIPTION
**Pull request by jsca-kwok**
This PR proposes a solution for checking whether an input is an anagram:
- Sort and compare

### CODE REVIEW

**Comment by jsca-kwok** [comment]
I would love any input for alternative solutions! :)

File: data_structures/arrays/strings/anagram_checker.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# How would you adapt your solution to such case?
+#
+# @param {String} s
+# @param {String} t
+# @return {Boolean}
+
+#
+# Approach 1: Sort and Compare
+#
+# Time Complexity: O(n log n)
```
Comment (vbrazo) [inline data_structures/arrays/strings/anagram_checker.rb:20]:
```suggestion
# Time Complexity: O(n log n). Assume that n is the length of s, sorting costs O(n log n), and comparing two strings costs O(n). Sorting time dominates and the overall time complexity is O(n log n).
```

File: data_structures/arrays/strings/anagram_checker.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#
```
Comment (vbrazo) [inline data_structures/arrays/strings/anagram_checker.rb:21]:
```suggestion
# Space Complexity: O(1). Space depends on the sorting implementation which, usually, costs O(1) auxiliary space if heapsort is used.
#
```

**Review by vbrazo — APPROVED** [review]

**Comment by vbrazo** [comment]
@jsca-kwok this can be solved using hash tables. I'll propose in a new PR.

**Pull request merged**